http-client/src/main/java/io/airlift/http/client/ResponseHandler.java

```diff
@@ -15,11 +15,23 @@
  */
 package io.airlift.http.client;
 
+import io.opentelemetry.api.trace.Span;
+
 public interface ResponseHandler<T, E extends Exception>
 {
     T handleException(Request request, Exception exception)
             throws E;
 
     T handle(Request request, Response response)
             throws E;
+
+    default void completeRequest(Runnable completer)
+    {
+        completer.run();
+    }
+
+    default void endSpan(Span span)
```

Comment:
why do we need this one?

Reply:
The span shouldn't end until all the bytes have been read

Reply:
Yeah, I figured